cmpserver/apiclient/clientset.go

```diff
@@ -11,13 +12,17 @@ import (
 	"google.golang.org/grpc"
 	"google.golang.org/grpc/credentials/insecure"
 
+	"github.com/argoproj/argo-cd/v2/common"
+	"github.com/argoproj/argo-cd/v2/util/env"
 	grpc_util "github.com/argoproj/argo-cd/v2/util/grpc"
 	"github.com/argoproj/argo-cd/v2/util/io"
 )
 
-const (
+var (
 	// MaxGRPCMessageSize contains max grpc message size
-	MaxGRPCMessageSize = 100 * 1024 * 1024
+	MaxGRPCMessageSize = env.ParseNumFromEnv(common.EnvArgoCDgRPCMaxSizeMB, 200, 0, math.MaxInt32) * 1024 * 1024
+	// MaxGRPCRetriesNum contains max grpc retries
+	MaxGRPCRetriesNum = env.ParseNumFromEnv(common.EnvArgoCDgRPCMaxRetries, 3, 0, math.MaxInt32)
```

Comment:
I'm somewhat against env vars which are loaded at the module level. I prefer when they're loaded by a command and then passed into the context where they're used. This has two advantages:

1) clarity for the user - they know precisely which commands can take advantage of this env var
2) clarity in the code - parameters are passed explicitly, so you know exactly what affects a given chain of function calls

I'd recommend implementing this by adding two new flags to the argocd-repo-server command and then loading these settings into a new `cmpServerClientSettings` struct. That struct should then be set on the repo server struct and passed into each function where it's used. It's verbose, but extremely clear.